editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -196,18 +196,72 @@ public boolean isProjectSpringBoot(IProject project) {
 		return CamelCatalogUtils.hasSpringBootDependency(projectDependencies);
 	}
 
+	
+	private org.fusesource.ide.camel.model.service.core.catalog.Dependency makeDependency(
+			String grpId, String artId, String version) {
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency dep = 
+				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
+		dep.setArtifactId(artId);
+		dep.setGroupId(grpId);
+		dep.setVersion(version);
+		return dep;
+	}
+	
 	/**
-	 * Add JAX-RS dependency
+	 * Add required dependencies
 	 */
 	private void updateDependencies(IProgressMonitor monitor) {
-		List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> deps = new ArrayList<>();
-		org.fusesource.ide.camel.model.service.core.catalog.Dependency one = 
-				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
-		one.setArtifactId("jboss-jaxrs-api_2.0_spec"); //$NON-NLS-1$
-		one.setGroupId("org.jboss.spec.javax.ws.rs"); //$NON-NLS-1$
-		one.setVersion("1.0.0.Final-redhat-1"); //$NON-NLS-1$
-		deps.add(one);
 		IProject project = options.getProject();
+
+		// use project to determine if we are building a spring or blueprint project
+		boolean isBlueprint = isProjectBlueprint(project);
+		boolean isSpringBoot = isProjectSpringBoot(project);
+		
+		// retrieve camel version for use with camel dependencies
+		CamelMavenUtils utils = new CamelMavenUtils();
+		String camelVersion = utils.getCamelVersionFromMaven(project, true);
+		
+		// now build a list of the dependencies we need
+		List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> deps = new ArrayList<>();
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency jaxrsSpec = 
+				makeDependency("org.jboss.spec.javax.ws.rs",  //$NON-NLS-1$
+						"jboss-jaxrs-api_2.0_spec",  //$NON-NLS-1$
+						"1.0.0.Final-redhat-1"); //$NON-NLS-1$
+		deps.add(jaxrsSpec);
+		
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency camelJackson = 
+				makeDependency("org.apache.camel",  //$NON-NLS-1$
+						"camel-jackson",  //$NON-NLS-1$
+						camelVersion);
+		deps.add(camelJackson);
+		
+		org.fusesource.ide.camel.model.service.core.catalog.Dependency camelCxf = 
+				makeDependency("org.apache.camel",  //$NON-NLS-1$
+						"camel-cxf",  //$NON-NLS-1$
+						camelVersion);
+		deps.add(camelCxf);
+
+		if (isBlueprint) {
+			org.fusesource.ide.camel.model.service.core.catalog.Dependency camelServlet = 
+					makeDependency("org.apache.camel",  //$NON-NLS-1$
+							"camel-servlet",  //$NON-NLS-1$
+							camelVersion);
+			deps.add(camelServlet);
+		} else {
+			// change this once we get the spring default component updated in wsdl2rest
```

Comment:
can you add a link the issue in wsdl2rest to track it more easily.
Maybe worth using a //TODO as there are better tooling to spot these kind of things.